test/test_view_ops.py

```diff
@@ -1101,6 +1101,14 @@ def test_reshape(self, device):
         self.assertEqual(x.data_ptr(), x.reshape_as(torch.rand(1, 9, 1)).data_ptr())
         self.assertRaises(RuntimeError, lambda: x.reshape_as(torch.rand(10, device=device)))
 
+        x = torch.rand((1, 4, 3, 5)).to(memory_format=torch.channels_last)
+        y = x.reshape(1, 4, 3, 5)
+        self.assertEqual(x, y)
```

Comment:
@CaoE Since your fix tries to make sure the channels last format can be propagated even if a channels_last tensor is reshaped before computing. How about check the propagation can work with your fix (I suppose it should fail without the fix)?

Reply:
> Is this testing the functionality this change is affecting or just adds a new test? 

Just adds a new test here.

> How about check the propagation can work with your fix (I suppose it should fail without the fix)?

Thank you for pointing me out. I added a new test in  `memory_format_test.cpp` to check if the propagation can work.